mcp/client.go

```diff
@@ -297,37 +311,51 @@ func (c *Client) createMessage(ctx context.Context, req *CreateMessageRequest) (
 
 func (c *Client) elicit(ctx context.Context, req *ElicitRequest) (*ElicitResult, error) {
 	if c.opts.ElicitationHandler == nil {
-		// TODO: wrap or annotate this error? Pick a standard code?
-		return nil, jsonrpc2.NewError(codeUnsupportedMethod, "client does not support elicitation")
-	}
-
-	// Validate that the requested schema only contains top-level properties without nesting
-	schema, err := validateElicitSchema(req.Params.RequestedSchema)
-	if err != nil {
-		return nil, jsonrpc2.NewError(codeInvalidParams, err.Error())
+		return nil, jsonrpc2.NewError(codeInvalidParams, "client does not support elicitation")
 	}
 
-	res, err := c.opts.ElicitationHandler(ctx, req)
-	if err != nil {
-		return nil, err
+	// Validate the elicitation parameters based on the mode.
+	mode := req.Params.Mode
+	if mode == "" {
+		mode = "form"
 	}
 
-	// Validate elicitation result content against requested schema
-	if schema != nil && res.Content != nil {
-		// TODO: is this the correct behavior if validation fails?
-		// It isn't the *server's* params that are invalid, so why would we return
-		// this code to the server?
-		resolved, err := schema.Resolve(nil)
+	switch mode {
+	case "form":
+		if req.Params.URL != "" {
+			return nil, jsonrpc2.NewError(codeInvalidParams, "URL must not be set for form elicitation")
+		}
+		schema, err := validateElicitSchema(req.Params.RequestedSchema)
 		if err != nil {
-			return nil, jsonrpc2.NewError(codeInvalidParams, fmt.Sprintf("failed to resolve requested schema: %v", err))
+			return nil, jsonrpc2.NewError(codeInvalidParams, err.Error())
 		}
-
-		if err := resolved.Validate(res.Content); err != nil {
-			return nil, jsonrpc2.NewError(codeInvalidParams, fmt.Sprintf("elicitation result content does not match requested schema: %v", err))
+		res, err := c.opts.ElicitationHandler(ctx, req)
+		if err != nil {
+			return nil, err
 		}
+		// Validate elicitation result content against requested schema.
+		if schema != nil && res.Content != nil {
+			resolved, err := schema.Resolve(nil)
+			if err != nil {
+				return nil, jsonrpc2.NewError(codeInvalidParams, fmt.Sprintf("failed to resolve requested schema: %v", err))
+			}
+			if err := resolved.Validate(res.Content); err != nil {
+				return nil, jsonrpc2.NewError(codeInvalidParams, fmt.Sprintf("elicitation result content does not match requested schema: %v", err))
+			}
+		}
+		return res, nil
+	case "url":
+		if req.Params.RequestedSchema != nil {
```

Comment:
double checking: do we need to add any documentation about what fields are valid for certain requests? (i wasn't really sure where this documentation should go, though, which is another sign this might not be needed. i wanted to double check)

Reply:
I'd documented this in the field documentation for the type. I think that's a good place for it; do you think that documentation should be improved?